timescaledb/tutorials/ingest-real-time-websocket-data.md

```diff
@@ -0,0 +1,346 @@
+    about the status of your connection:
+    ```bash
+    {'event': 'subscribe-status',
+     'status': 'ok',
+     'success': [
+            {'symbol': 'BTC/USD', 'exchange': 'Coinbase Pro', 'mic_code': 'Coinbase Pro', 'country': '', 'type': 'Digital Currency'},
+            {'symbol': 'ETH/USD', 'exchange': 'Huobi', 'mic_code': 'Huobi', 'country': '', 'type': 'Digital Currency'}
+        ],
+     'fails': None
+    }
+    ```
+
+</procedure>
+
+When you have established a connection to the websocket server, 
+wait a few seconds, and you can see actual data records, like this:
+
+```bash
+{'event': 'price', 'symbol': 'BTC/USD', 'currency_base': 'Bitcoin', 'currency_quote': 'US Dollar', 'exchange': 'Coinbase Pro', 'type': 'Digital Currency', 'timestamp': 1652438893, 'price': 30361.2, 'bid': 30361.2, 'ask': 30361.2, 'day_volume': 49153}
+{'event': 'price', 'symbol': 'BTC/USD', 'currency_base': 'Bitcoin', 'currency_quote': 'US Dollar', 'exchange': 'Coinbase Pro', 'type': 'Digital Currency', 'timestamp': 1652438896, 'price': 30380.6, 'bid': 30380.6, 'ask': 30380.6, 'day_volume': 49157}
+{'event': 'heartbeat', 'status': 'ok'}
+{'event': 'price', 'symbol': 'ETH/USD', 'currency_base': 'Ethereum', 'currency_quote': 'US Dollar', 'exchange': 'Huobi', 'type': 'Digital Currency', 'timestamp': 1652438899, 'price': 2089.07, 'bid': 2089.02, 'ask': 2089.03, 'day_volume': 193818}
+{'event': 'price', 'symbol': 'BTC/USD', 'currency_base': 'Bitcoin', 'currency_quote': 'US Dollar', 'exchange': 'Coinbase Pro', 'type': 'Digital Currency', 'timestamp': 1652438900, 'price': 30346.0, 'bid': 30346.0, 'ask': 30346.0, 'day_volume': 49167}
+```
+
+Each price event gives you multiple data points about the given trading pair
+such as the name of the exchange, and the current price. You can also
+occasionally see `heartbeat` events in the response; these events signal
+the health of the connection over time.
+
+At this point the websocket connection works and data keeps flowing. You need
+to implement the `on_event` function so data gets ingested into TimescaleDB.
+
+## Ingesting websocket data into TimescaleDB
+Now that the websocket connection is set up, you can use the `on_event` function
+to ingest data into the database. 
+
+Usually, when you ingest data into a transactional database, ingesting a thousand 
+records in one transaction is faster than ingesting them in a thousand transactions.
+Because TimescaleDB is a transactional database, it creates a new transaction
+whenever you try to insert a new record. This means that it's better to insert 
+data records in a batch, so you can ingest multiple records in a single transaction.
+This approach means you can achieve higher ingest rates than if you were to
+insert data records one-by-one.
+
+When you insert data in batches, as opposed to one record at a time, the
+database spends less time managing the connection, and transactions,  so
+ingestion becomes faster overall. 
```

Comment:
```suggestion
When you ingest data into a transactional database like TimescaleDB, it is more efficient to
insert data in batches rather than inserting data row-by-row. Using one transaction to
insert multiple rows can significantly increase the overall ingest capacity and speed
of your TimescaleDB instance.
```

I felt like there was a lot of repetition here. If I've trimmed it down too much, feel free to modify a bit. But, don't say things like "usually" when, for all intents, it is true. We always suggest applications insert in batches. I literally can't think of a situation where it would be faster. It might be easier, but overall not faster.